Source/Particles/Collision/BinaryCollision/Coulomb/ElasticCollisionPerez.H

```diff
@@ -82,18 +83,23 @@ void ElasticCollisionPerez (
     T_PR n1  = T_PR(0.0);
     T_PR n2  = T_PR(0.0);
     T_PR n12 = T_PR(0.0);
-    for (int i1=I1s; i1<static_cast<int>(I1e); ++i1) { n1 += w1[ I1[i1] ]; }
-    for (int i2=I2s; i2<static_cast<int>(I2e); ++i2) { n2 += w2[ I2[i2] ]; }
-    n1 = n1 / dV; n2 = n2 / dV;
-    {
-      int i1 = I1s; int i2 = I2s;
-      for (int k = 0; k < amrex::max(NI1,NI2); ++k)
-      {
-        n12 += amrex::min( w1[ I1[i1] ], w2[ I2[i2] ] );
-        ++i1; if ( i1 == static_cast<int>(I1e) ) { i1 = I1s; }
-        ++i2; if ( i2 == static_cast<int>(I2e) ) { i2 = I2s; }
-      }
-      n12 = n12 / dV;
+    if ( isSameSpecies ) {
+        for (int i1=I1s; i1<static_cast<int>(I1e); ++i1) { n1 += w1[ I1[i1] ]; }
+        for (int i2=I2s; i2<static_cast<int>(I2e); ++i2) { n1 += w2[ I2[i2] ]; }
+        n1 = n1 / dV; n2 = n1; n12 = n1;
```

Comment:
I agree with the calculation of `n1` and `n2` here, but I think that the calculation of `n12` is too simple (i.e. only valid in the case where all weights are equal). I think that we should instead use the same code that calculates `n12` below.